### PR TITLE
[fix]: [CDS-71280]: increasing page size for ECR image tags fetch

### PR DESCRIPTION
**Pull request by markoharness**
## Describe your changes

Updated the page size from default value(100) to 1000 for fetching images from ECR.
This will reduce time for fetching images (tags) from ECR

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49159)

### CODE REVIEW

**Comment by markoharness** [comment]
trigger pmd

**Comment by markoharness** [comment]
trigger uts

**Comment by markoharness** [comment]
trigger uts

**Comment by markoharness** [comment]
trigger uts

**Comment by yogesh-chauhan** [comment]
could you please update the PR description as well ?

**Review by arvind-choudhary-h — APPROVED** [review]
> Reviewed 3 of 4 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49159)* status: 4 file(s) to review. 1 of 2 approvals obtained.

File: 960-api-services/src/main/java/software/wings/helpers/ext/ecr/EcrService.java [hunk 1]
```diff
@@ -21,7 +21,7 @@
  */
 @OwnedBy(CDC)
 public interface EcrService {
-  int MAX_NO_OF_TAGS_PER_IMAGE = 10000;
+  int MAX_NO_OF_IMAGES = 1000;
```
Comment (vlica-harness) [inline 960-api-services/src/main/java/software/wings/helpers/ext/ecr/EcrService.java:24]: minor type?: should it be MAX_NR_OF_IMAGES ? NR vs NO

File: 960-api-services/src/main/java/software/wings/helpers/ext/ecr/EcrServiceImpl.java [hunk 2]
```diff
@@ -127,11 +133,12 @@ public List<BuildDetailsInternal> getBuilds(AwsInternalConfig awsConfig, String
               });
             });
         describeImagesRequest.setNextToken(describeImagesResult.getNextToken());
+        log.debug("Finished processing page no. {} for describeImages API call", pageCounter);
       } while (describeImagesRequest.getNextToken() != null);
     } catch (Exception e) {
       return getBuildsFallback(awsConfig, registryId, imageUrl, region, imageName);
     }
-    // Sorting at build tag for docker artifacts.
+    log.debug("GetBuilds describeImages API has done with the fetch of {} tags", buildDetailsInternals.size());
```
Comment (vlica-harness) [inline 960-api-services/src/main/java/software/wings/helpers/ext/ecr/EcrServiceImpl.java:141]: is debug log intentionally or should be as info?

**Review by vlica-harness — APPROVED** [review]

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/ecr/EcrArtifactTaskHandler.java [hunk 3]
```diff
@@ -57,7 +57,7 @@ public ArtifactTaskExecutionResponse getBuilds(EcrArtifactDelegateRequest attrib
     String ecrimageUrl = awsEcrApiHelperServiceDelegate.getEcrImageUrl(awsInternalConfig,
         attributesRequest.getRegistryId(), attributesRequest.getRegion(), attributesRequest.getImagePath());
     builds = ecrService.getBuilds(awsInternalConfig, attributesRequest.getRegistryId(), ecrimageUrl,
-        attributesRequest.getRegion(), attributesRequest.getImagePath(), MAX_NO_OF_TAGS_PER_IMAGE);
```
Comment (yogesh-chauhan) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/ecr/EcrArtifactTaskHandler.java:60]: initially the value was 10K and now we have reduced it to 1K ?

Comment (markoharness) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/ecr/EcrArtifactTaskHandler.java:60]: Yes, but that variable was not used anywhere. Now it is being used.

File: 960-api-services/src/main/java/software/wings/helpers/ext/ecr/EcrServiceImpl.java (hunk 2 above)
Comment (markoharness) [reply 960-api-services/src/main/java/software/wings/helpers/ext/ecr/EcrServiceImpl.java:141]: Yes, debug logs lines are intentional.

**Comment by markoharness** [comment]
trigger messagecheck

**Review by vlica-harness — APPROVED** [review]

**Review by arvind-choudhary-h — APPROVED** [review]
> Reviewed 1 of 1 files at r3, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49159)* status: 7 file(s) to review. 2. unresolved discussions.

**Comment by markoharness** [comment]
trigger review

**Pull request merged**